### DISCUSSION TITLE
Not able to get static file (font .ttf files) in vercel deployment

### DISCUSSION THREAD

**Discussion by SawkaDev** [vercel/community#1240] [Help] [answered]
Hi,

I am trying to generate a PDF on server side using PDFMake library. Everything works fine on localhost, but not when i deploy to vercel. 

Error Encountered: `{"errno":-2,"syscall":"open","code":"ENOENT","path":"/assets/fonts/Roboto-Bold.ttf"}`

Seems vercel can not locate the file I need to use from the serverside. 

Here is the directory structure in localhost:
![image](https://user-images.githubusercontent.com/19200789/211417194-e3319d2f-9d4b-4b1a-a22d-8f458ca2d3f7.png)

Here is the directory structure when i check the "source" tab in the vercel deployment:
![image](https://user-images.githubusercontent.com/19200789/211417251-dde4bed6-5c31-48a9-b534-83f89a1074cc.png)

Any recommendations on how to debug / fix this issue?

Thanks!

**Comment by amyegan** [comment, marked as answer]
Hi @Sliffcak! Do you have a public repo or [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to get a better look at the problem?

You might just need to adjust the resource path, but I'm not sure what to suggest without knowing more about your project. The guide on [using files in Serverless Functions](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions) might help.

**Reply by SawkaDev** [reply]
@amyegan thanks for the reply. If I try the method as shown in the link you sent by using the path.join function I still get this error

`{"errno":-2,"syscall":"open","code":"ENOENT","path":"/var/task/assets/fonts/Roboto-Bold.ttf"}`


seems the public folder gets uploaded to `/var/task` but still getting ENOENT error. Ill work on creating a minimal reproduction. Will update soon!

**Reply by amyegan** [reply]
Does it work if you put `/public` in front of `/assets/fonts` in the path?

**Reply by SawkaDev** [reply]
@amyegan sadly not. Again thank you for all your help. 

Here is a minimal reproduction:  https://github.com/Sliffcak/minimal-public-asset-not-found

I made it very simple and it only involves trying to access a file in the public folder.

Steps for reproduction localhost:

1. Clone
2. npm install
3. npm run dev
4. Open localhost:3000 and click any of the buttons on the homescreen (the buttons call an API to the backend, the backend is supposed to be able to access a file stores in the `/public/staticAssets` folder and then send back the size of that file to the front end to display the size)
5. On localhost it will report the size properly to the front end for both the image file and also the font file

Reproduction Steps on Production

1. Deploy the app to vercel
2. open website of deployment
3. in vercel go to "Functions" tab so you can view the realtime logs / error
4. Click any of the buttons
5. you should see a print similar to `{"errno":-2,"syscall":"stat","code":"ENOENT","path":"staticAssets/Roboto-Regular.ttf"}`

I have tried all sorts of directory structures for the serverless function to try and access the file but I always get ENOENT. Or if you remove the try, catch block in the API handler you will get the ERROR of "MODULE_NOT_FOUND".

Let me know if this is what you need or any more information!

**Reply by amyegan** [reply]
I was able to get it working by making a couple of changes to the file paths. Opened a new PR to show the change https://github.com/Sliffcak/minimal-public-asset-not-found/pull/1 and you can see my test deployment at [minimal-public-asset-not-found-amyegan.vercel.app](https://minimal-public-asset-not-found-amyegan.vercel.app)

**Reply by SawkaDev** [reply]
@amyegan I am at a loss of words. I spent so many hours on such a simple issue! Feeling really dumb for missing such an easy concept. I can not thank you enough.

I think looking at the deployment source was messing me up the most, since I did not see a public folder in the development source.

Thank you so much! All fixed now in my minimal example and my real app!

**Comment by SawkaDev** [comment]
Answer for anyone who wants to figure out how to access a file stores in the /public folder in vercel deployment. Just do this

If your file is in `/public/assets` you just get it simply by the following:

`path.join(process.cwd(),"public","assets",FILENAME);`

**Reply by aly-and-u** [reply]
Can you send the code here please?

**Reply by amyegan** [reply]
@aly-and-u, you can find some example code in [this guide about using files in serverless functions](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions)
